### PR TITLE
[Merged by Bors] - Execute persistent SmartModule

### PR DESCRIPTION
**Pull request by nacardin**
To resolve https://github.com/infinyon/fluvio/issues/1774
To resolve https://github.com/infinyon/fluvio/issues/1791

### CODE REVIEW

**Comment by nacardin** [comment]
Added tests

File: crates/fluvio/src/consumer.rs [hunk 1]
```diff
@@ -429,28 +431,31 @@ impl PartitionConsumer {
...
-                debug!("Using WASM V1 API");
-                let wasm = module.wasm.get_raw()?;
-                stream_request.wasm_module = wasm.into_owned();
-            } else {
-                // SmartStream V2
-                debug!("Using WASM V2 API");
-                if stream_fetch_version < GZIP_WASM_API as i16 {
-                    module.wasm.to_raw()?;
+        if let Some(smart_module) = config.smart_module {
+            println!("vers {:?}", stream_fetch_version);
```
Comment (sehz) [inline crates/fluvio/src/consumer.rs:435]: change to debug like: `debug!(stream_version_version,"version")`

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  Nice job on test refactoring. 1 minor change

File: crates/fluvio/src/consumer.rs (hunk 1 above)
Comment (nacardin) [reply crates/fluvio/src/consumer.rs:435]: good catch

**Review by sehz — APPROVED** [review]
Nice job!

**Pull request closed without merging**